src/Lecture4.hs

```diff
@@ -166,25 +200,38 @@ lines.
 -}
 
 data Stats = Stats
-    { statsTotalPositions :: Sum Int
-    , statsTotalSum       :: Sum Int
-    , statsAbsoluteMax    :: Max Int
-    , statsAbsoluteMin    :: Min Int
-    , statsSellMax        :: Maybe (Max Int)
-    , statsSellMin        :: Maybe (Min Int)
-    , statsBuyMax         :: Maybe (Max Int)
-    , statsBuyMin         :: Maybe (Min Int)
-    , statsLongest        :: MaxLen
-    } deriving (Show, Eq)
+  { statsTotalPositions :: Sum Int,
+    statsTotalSum :: Sum Int,
+    statsAbsoluteMax :: Max Int,
+    statsAbsoluteMin :: Min Int,
+    statsSellMax :: Maybe (Max Int),
+    statsSellMin :: Maybe (Min Int),
+    statsBuyMax :: Maybe (Max Int),
+    statsBuyMin :: Maybe (Min Int),
+    statsLongest :: MaxLen
+  }
+  deriving (Show, Eq)
 
 {-
 The 'Stats' data type has multiple fields. All these fields have
 'Semigroup' instances. This means that we can implement a 'Semigroup'
 instance for the 'Stats' type itself.
 -}
 
+accMaybe :: Semigroup a => Maybe a -> Maybe a -> Maybe a
+accMaybe m Nothing = m
+accMaybe Nothing m = m
+accMaybe (Just !m) (Just !n) = Just (m <> n)
```

Comment:
That's absolutely correct implementation 👍🏻 
You can write it a bit shorter though by pattern matching only on the final result:

```suggestion
accMaybe a b = case a <> b of
    Nothing -> Nothing
    Just !x -> Just x
```